homeassistant/components/legrandinone.py

```diff
@@ -0,0 +1,505 @@
+            self.hass.bus.async_fire(EVENT_BUTTON_PRESSED, {
+                ATTR_ENTITY_ID: self.entity_id,
+                ATTR_STATE: event.get(EVENT_TYPE_COMMAND),
+            })
+            _LOGGER.debug("Fired bus event for %s: %s",
+                          self.entity_id, event.get(EVENT_TYPE_COMMAND))
+
+    def _handle_event(self, event):
+        """Platform specific event handler."""
+        raise NotImplementedError()
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def name(self):
+        """Return a name for the device."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        if self.assumed_state:
+            return False
+        return self._state
+
+    @property
+    def assumed_state(self):
+        """Assume device state until first device event sets state."""
+        return self._state is None
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    @callback
+    def _availability_callback(self, availability):
+        """Update availability state."""
+        self._available = availability
+        self.async_schedule_update_ha_state()
+
+    async def async_added_to_hass(self):
+        """Register update callback."""
+        # Remove temporary bogus entity_id if added
+#         tmp_entity = TMP_ENTITY.format(self.legrand_id)
+#         if tmp_entity in self.hass.data[DATA_ENTITY_LOOKUP][
+#                 EVENT_KEY_SENSOR][self.legrand_id]:
+#             self.hass.data[DATA_ENTITY_LOOKUP][
+#                 EVENT_KEY_SENSOR][self.legrand_id].remove(tmp_entity)
+
+        # Register id and aliases
+        self.hass.data[DATA_ENTITY_LOOKUP][
+            EVENT_KEY_COMMAND][self.legrand_id].append(self.legrand_id)
+
+        async_dispatcher_connect(self.hass, SIGNAL_AVAILABILITY,
+                                 self._availability_callback)
+        async_dispatcher_connect(self.hass,
+                                 SIGNAL_HANDLE_EVENT.format(self.legrand_id),
+                                 self.handle_event_callback)
+
+        # Process the initial event now that the entity is created
+        if self._initial_event:
+            self.handle_event_callback(self._initial_event)
+
+
+class LegrandInOneCommand(LegrandInOneDevice):
+    """Singleton class to make IOBL command interface available to entities.
+
+    This class is to be inherited by every Entity class that is actionable
+    (switches/lights). It exposes the IOBL command interface for these
+    entities.
+
+    The IOBL interface is managed as a class level and set during setup (and
+    reset on reconnect).
+    """
+
+    _protocol = None
+
+    @classmethod
+    def set_iobl_protocol(cls, protocol):
+        """Set the IOBL asyncio protocol as a class variable."""
+        cls._protocol = protocol
```

Comment:
Refactor this and don't set the protocol on the class. Pass it to the entity when instantiating that via hass.data.

Reply:
I'm afraid I'm quite new to both python & home assistant. Could you provide some more guidance here ?

Reply:
`hass.data` is a dict that we can store data in that we need to share between eg component and platforms. Put the protocol there from the component and then fetch it in platform setup and give to each entity when creating those.

Reply:
I'm pushing a new version. Is it better like that?